xarray/tests/test_dataset.py

```diff
@@ -2871,6 +2871,26 @@ def test_to_and_from_dict(self):
                                      "without the key 'dims'"):
             Dataset.from_dict(d)
 
+    def test_to_and_from_dict_with_zero_dim(self):
+        ds = Dataset(OrderedDict([('a', ('t', np.array([], dtype='float64'))),
+                                  ('b', ('t', np.array([], dtype='float64'))),
+                                  ('t', ('t', np.array([], dtype='U1')))]))
+        expected = {'coords': {'t': {'dims': ('t',),
+                                     'data': [],
+                                     'attrs': {}}},
+                    'attrs': {},
+                    'dims': {'t': 0},
+                    'data_vars': {'a': {'dims': ('t',),
+                                        'data': [],
+                                        'attrs': {}},
+                                  'b': {'dims': ('t',),
+                                        'data': [],
+                                        'attrs': {}}}}
+
+        actual = ds.to_dict()
+        self.assertEqual(expected, actual)
+        self.assertDatasetIdentical(ds, Dataset.from_dict(actual))
```

Comment:
We're trying to move away from using the UnitTest assert methods. Instead, try this:

```Python
assert actual == expected
assert_identical(ds, Dataset.from_dict(actual))
```